lib/bunyan.js

```diff
@@ -1339,7 +1380,13 @@ RotatingFileStream.prototype.write = function write(s) {
         this.rotQueue.push(s);
         return false;
     } else {
-        return this.stream.write(s);
+        if ((this.alreadyWritten + this.stream.bytesWritten) > this.rotateFileSize) {
```

Comment:
Thanks for catching this!  Fixed!

On Sat, Feb 6, 2016 at 10:55 AM, Jim Tupper <notifications@github.com>
wrote:

> In lib/bunyan.js
> <https://github.com/trentm/node-bunyan/pull/288#discussion_r52103575>:
>
> > @@ -1339,7 +1380,13 @@ RotatingFileStream.prototype.write = function write(s) {
> >          this.rotQueue.push(s);
> >          return false;
> >      } else {
> > -        return this.stream.write(s);
> > +        if ((this.alreadyWritten + this.stream.bytesWritten) > this.rotateFileSize) {
>
> By checking the file and rotating immediately, we lose the log message
> that caused the rotation
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/trentm/node-bunyan/pull/288/files#r52103575>.
>



-- 
Kevin Smathers
Product Development Engineer
Tanium, Inc.
kevin.smathers@tanium.com